### PR TITLE
Allow excluding default values for emailProvider with `AUTH0_EXCLUDED_DEFAULTS`

### PR DESCRIPTION
**Pull request by sobil**
## ✏️ Changes

Exporting default values can be valuable for discovering the format and schema of the format. It can however lead to unintentionally setting configuration with incorrect values in an already running environment.

This change adds a mechanism to prevent the default values being included in the configuration dump

It also changes the output for the default values to be useful with Environment Variables or AUTH0_KEYWORD_REPLACE_MAPPINGS.

## 🔗 References

Main Issue: https://github.com/auth0/auth0-deploy-cli/issues/235

Example of issue occurring for an Auth0 customer: https://support.auth0.com/tickets/00455714
Potentially part of a solution to resolve: https://github.com/auth0/auth0-deploy-cli/issues/185

## 🎯 Testing

✅ This change has unit test coverage

✅ This change has integration test coverage

✅ This change has been tested for performance

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- Allow exluding default values for emailProvider with `AUTH0_EXCLUDED_DEFAULTS`
```
Comment (trevorgk) [inline CHANGELOG.md:10]: typo: 
```suggestion
- Allow excluding default values for emailProvider with `AUTH0_EXCLUDED_DEFAULTS`
```

Comment (trevorgk) [reply CHANGELOG.md:10]: Also in the PR title

**Comment by shushen** [comment]
> code should throw an error if the user is including explicitly values in INCLUDED_PROPS but these values are being excluded by AUTH0_EXCLUDED_DEFAULTS so the user knows that is an error with the configuration and the result wouldn't be as expected.

@luisbritos IMO this should not be required as there is no conflict.

- `AUTH0_EXCLUDED_DEFAULTS` specified the behaviour for the handlers to fill in defaults if the fields are not present from results returned from management API. This happens before the [`clean_asset()`](https://github.com/auth0/auth0-deploy-cli/blob/9d8a0b3d26cb276c51233d1bb00d9fb5a6933b00/src/context/yaml/index.js#L105) function is called to process `INCLUDED_PROPS` and `EXCLUDED_PROPS`. 
- `INCLUDED_PROPS` only applies when making an exception to include [properties that are usually stripped off](https://github.com/auth0/auth0-deploy-cli/blob/9d8a0b3d26cb276c51233d1bb00d9fb5a6933b00/src/readonly.js#L4)

Therefore, 
- if a field is specified in `EXCLUDED_PROPS`, it will be excluded no matter how `AUTH0_EXCLUDED_DEFAULTS` is configured, and
- if `emailProvider` is specified in `AUTH0_EXCLUDED_DEFAULTS`, or for any property that we might want to fill in with defaults in future, it's unlikely we will include them as part of the [properties to be stripped off](https://github.com/auth0/auth0-deploy-cli/blob/9d8a0b3d26cb276c51233d1bb00d9fb5a6933b00/src/readonly.js#L4), but in case we do, then they can still be controlled by `INCLUDED_PROPS`, independently of whether we fill them with defaults or not.

**Comment by shushen** [comment]
@luisbritos I'm merging this PR - we can address any further concerns separately.

@sobil Many thanks for the awesome PR!

**Pull request merged**